sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -220,7 +220,7 @@ object PullupCorrelatedPredicates extends Rule[LogicalPlan] with PredicateHelper
     */
```

Comment:
Changes in this file are not quite necessary, but just to match the code in the analyzer side: when we need to pass around an outer plan, just pass it instead of its children.